src/vision/status_quo/barbara_anguishes_over_http.md

```diff
@@ -0,0 +1,47 @@
+
+As they are pair programming, they get the part of the library where HTTP will be involved and Alan asks Barbara, "OK, how do I make an HTTP request?".
+
+As an experienced async Rust developer Barbara has been dreading this question from the start of the project. She's tempted to ask "How long do you have?", but she quickly gathers herself and sstarts to outline the various considerations. She starts with a relatively simple question: "Should we use an HTTP library with a sync interface or an async interface?".
+
+Alan, somewhat confused by the question since this isn't a concern in the languages he's used to, replies with his own question: "I don't know, what do you think?". Barbara thinks for a second. She knows that bridging sync and async can be difficult, and so there's another question they need to answer first: "Well, are we going to expose a sync or an async interface to the users of our library?".
+
+Alan, now feeling completely out of his element, replies as confident as he can: "Everybody wants to use async these days. Let's do that.". He braces for Barbara's answer as he's not even sure what he said is actually true.
+
+Barbara replies, "If we expose async API then we need to decide which async HTTP implementation we we will use". As she finishes saying this, Barbara feels slightly uneasy. She knows that it is possible to use a sync HTTP library and expose it through an async API, but she fears totally confusing Alan and so decides to not mention this fact.
+
+Barbara looks over at Alan and sees a blank stare on his face. She repeats the question: "So, which async HTTP implementation should we use?". Alan responds with the only thing that comes to his mind: "which one is the best?" to which Barbara responds "Well, it depends on which async runtime you're using". 
+
+* Alan, feeling utterly dejected and hoping that the considerations will soon end tries a new route out of this conversation: "Can we allow the user of the library to decide?". 
+
+Barbara thinks to herself, "Oh boy, we could provide a trait that abstracts over the HTTP request and response and allow the user to provide the implementation for whatever HTTP library they want... BUT, if we ever need any additional functionality that an async runtime needs to expose - like async locks or async timers - we might be forced to pick an actual runtime implementation on behalf of the user... Perhaps, we can put the most popular runtime implementations behind feature flags and let the user chose that way... BUT what if we want to allow plugging in of different runtimes?"
+
+Alan, having watched Barbara stare off into the distance for what felt like a half-hour, feels bad for his colleague. All he can think to himself is how Rust is so much more complicated that C#.
+
+## 🤔 Frequently Asked Questions
+
+* **What are the morals of the story?**
```

Comment:
The style of the FAQ sections was changed in #118 / #119, using sub headings (###) instead of nested bullet lists.